app/Http/Controllers/SearchController.php

```diff
@@ -100,15 +107,23 @@ function (Builder $query) use ($id, $escaped_term) {
 		if ($photos != null) {
 			$i = 0;
 			foreach ($photos as $photo) {
-				$return['photos'][$i] = $photo->prepareData();
-				++$i;
+				if (Session::has('UserID') || $photo->get_public()) {
+					$return['photos'][$i] = $photo->prepareData();
+					++$i;
+				}
 			}
 		}
 
 		/**
 		 * Albums
 		 */
-		$query = Album::where('owner_id', '=', $id);
+		$query = Album::where('id', '>', 0);
```

Comment:
Some question as about Photo query above.

Reply:
> PS: I do not like query builders too much ;-)

Sorry... the WHERE-brackets again, see above.